src/hooks/useWalletConnection.ts

```diff
@@ -0,0 +1,16 @@
+import { walletApi } from 'api'
+import { useState, useEffect } from 'react'
+import { WalletInfo } from 'types'
+
+export const useWalletConnection = (): WalletInfo => {
+  const [walletInfo, setWalletInfo] = useState<WalletInfo>({ isConnected: false })
+
+  useEffect((): (() => void) => {
+    walletApi.addOnChangeWalletInfo(setWalletInfo, true)
+
+    return function cleanup(): void {
+      walletApi.removeOnChangeWalletInfo(setWalletInfo)
```

Comment:
this whole passage could just be
```
useEffect(() => walletApi.addOnChangeWalletInfo(setWalletInfo), [])
```
if you implement my suggestion for `WalletApi::addOnChangeWalletInfo`

Reply:
yep, I just read this now after implementing it for your previous comment.

I did that, but just not in one line. 
Usually when the line does several things, I prefer to separate them so each line has it’s porpouse:
![image](https://user-images.githubusercontent.com/2352112/66036913-b45ff080-e50e-11e9-9716-57addf37037b.png)

1. useEffect line + definition of the argument function
2. Impl of the effect
3. dependencies